scripts/daily_upload_s3.coffee

```diff
@@ -0,0 +1,49 @@
+knox = require 'knox'
+Q = require 'bluebird-q'
+mongojs = require 'mongojs'
+csv = require 'fast-csv'
+fs = require 'fs'
+path = require 'path'
+moment = require 'moment'
+
+# Setup environment variables
+env = require 'node-env-file'
+switch process.env.NODE_ENV
+  when 'test' then env path.resolve __dirname, '../.env.test'
+  when 'production', 'staging' then ''
+  else env path.resolve __dirname, '../.env'
+
+# Connect to database
+db = mongojs(process.env.MONGOHQ_URL, ['articles'])
+
+# Setup CSV
+filename = "export_" + moment().format('YYYYMMDDhhmmss') + ".csv"
+csvStream = csv.createWriteStream { headers: true }
+writableStream = fs.createWriteStream('scripts/tmp/' + filename)
+
+csvStream.pipe(writableStream)
+csvStream.write(["id", "author_id", "auction_id", "contributing_authors", "fair_id", "featured", "featured_artist_ids", "featured_artwork_ids", "partner_ids", "primary_featured_artist_ids", "slugs", "tags", "title", "tier", "published_at","show_ids","section_ids","thumbnail_image","thumbnail_title"])
+
+db.articles.find({ published: true })
+  .on('data', (doc) ->
+    if doc
+      published_at = if doc.published_at then moment(doc.published_at).format('YYYYMMDDhhmmss') + " EST" else ''
+      csvStream.write([doc._id, doc.author_id, doc.auction_id, doc.contributing_authors, doc.fair_id, doc.featured, doc.featured_artist_ids, doc.featured_artwork_ids, doc.partner_ids, doc.primary_featured_artist_ids, doc.slugs, doc.tags, doc.title, doc.tier, published_at, doc.show_ids, doc.section_ids, doc.thumbnail_image, doc.thumbnail_title])
+  ).on 'end', ->
+
+    # End Streaming
+    csvStream.end()
+
+    # Setup S3 Client
+    client = knox.createClient
+      key: process.env.S3_KEY
+      secret: process.env.S3_SECRET
+      bucket: process.env.FULCRUM_BUCKET
+
+    client.putFile 'scripts/tmp/' + filename, "reports/positron_articles/#{filename}", {
```

Comment:
Writing to a local file on disk and then uploading that file all at once is legit, but if you're curious you could probably avoid the extra local writing and reading by piping your `csvStream` directly to knox's put API (related https://github.com/Automattic/knox#examples). e.g. something like this:

```coffeescript
s3PutStream = client.put('scripts/tmp/' + filename)
csvStream.pipe(s3PutStream)
```